services/queue/src/utils.js

```diff
@@ -113,8 +113,19 @@ const artifactUtils = {
           }
         }
       };
-      const deleteSingleObject = async (bucket, entry) =>
-        bucket.deleteObject(entry.details.prefix);
+      const deleteSingleObject = async (bucket, entry) => {
+        try {
+          return await bucket.deleteObject(entry.details.prefix);
+        } catch (err) {
+          if (err.name === 'NoSuchKey' || err.code === 'NoSuchKey' || /NoSuchKey/.match(err.message)) {
```

Comment:
So here's GCS documentation: https://cloud.google.com/storage/docs/xml-api/delete-object 
it will return `404`/`NoSuchKey` if object not found

Amazon S3 https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html#API_DeleteObject_ResponseSyntax is sending `204` always

Maybe it's not good to check different props like this, and instead join strings or stringifying whole object would make just one check

Reply:
this is also coming from service logs and error console in GCP